src/string/escapeRegExp.zig

```diff
@@ -0,0 +1,76 @@
+const special_characters = "|\\{}()[]^$+*?.-";
```

Comment:
would love if this had more tests or we pulled from jsc's `RegExp.escape` implementation.